dds/DCPS/RemoveAssociationSweeper.h

```diff
@@ -111,7 +111,10 @@ RemoveAssociationSweeper<T>::RemoveAssociationSweeper(ACE_Reactor* reactor,
 
 template <typename T>
 RemoveAssociationSweeper<T>::~RemoveAssociationSweeper()
-{ }
+{
+  ACE_Guard<ACE_Thread_Mutex> guard(mutex_);
+  info_set_.clear();
```

Comment:
Since the `RemoveAssociationSweeper` object is being destroyed, this doesn't seem like it's necessary.
Being a reference-counted object, no reference to this object exists when the destructor starts running.
If this isn't true in practice then there's a bug somewhere else.